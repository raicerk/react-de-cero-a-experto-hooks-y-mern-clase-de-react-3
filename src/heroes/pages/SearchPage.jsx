import { useLocation, useNavigate } from "react-router-dom";
import { useForm } from "../../hooks/useform";
import { HeroCard } from "../components/HeroCard";
import queryString from "query-string";
import { getHeroesByName } from "../helpers/getHeroesByName";

export const SearchPage = () => {
  const navigate = useNavigate();
  const location = useLocation();

  const { q = "" } = queryString.parse(location.search);

  const heroes = getHeroesByName(q);

  const { searchText, onInputChange } = useForm({ searchText: q });

  const onSearchSubmit = (event) => {
    event.preventDefault();
    navigate(`?q=${searchText}`);
  };

  return (
    <>
      <h1>Search</h1>
      <hr />
      <div className="row">
        <div className="col-5">
          <h4>Searching</h4>
          <hr />
          <form onSubmit={onSearchSubmit}>
            <input
              type="text"
              placeholder="Search a hero"
              className="form-control"
              name="searchText"
              autoComplete="off"
              value={searchText}
              onChange={onInputChange}
            />
            <button className="btn btn-outline-primary" type="submit">
              Search
            </button>
          </form>
        </div>
        <div className="col-7">
          <h4>Results</h4>
          <hr />

          {q === "" ? (
            <div className="alert alert-primary">Search a hero:</div>
          ) : (
            heroes.length === 0 && (
              <div className="alert alert-danger">
                No hero with <b>{q}</b>
              </div>
            )
          )}

          {heroes.map((hero) => {
            return <HeroCard key={hero.id} {...hero} />;
          })}
        </div>
      </div>
    </>
  );
};
