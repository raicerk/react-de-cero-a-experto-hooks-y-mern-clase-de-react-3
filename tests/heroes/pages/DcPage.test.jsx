import { render, screen } from "@testing-library/react";
import { DcPage } from "../../../src/heroes/pages/DcPage";
import { MemoryRouter } from "react-router-dom";

describe("Test de <DcPage/> component", () => {
  test("Debe mostrarse el componente por defecto", () => {
    const { container } = render(
      <MemoryRouter>
        <DcPage />
      </MemoryRouter>
    );
    expect(screen.findAllByText("DC")).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
});
