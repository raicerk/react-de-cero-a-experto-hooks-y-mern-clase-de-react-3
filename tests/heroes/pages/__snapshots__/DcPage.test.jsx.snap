// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test de <DcPage/> component Debe mostrarse el componente por defecto 1`] = `
<div>
  <h1>
    DC
  </h1>
  <hr />
  <div
    class="row rows-cols-1 row-cols-md-3 g-3"
  >
    <div
      class="col"
    >
      <div
        class="card"
      >
        <div
          class="row no-gutters"
        >
          <div
            class="col-4"
          >
            <img
              alt="Batman"
              class="card-img"
              src="/assets/dc-batman.jpg"
            />
          </div>
          <div
            class="col-8"
          >
            <div
              class="card-body"
            >
              <h5
                class="card-title"
              >
                Batman
              </h5>
              <p
                class="card-text"
              >
                Bruce Wayne
              </p>
              <p
                class="card-text"
              >
                <small
                  class="text-muted"
                >
                  Detective Comics #27
                </small>
              </p>
              <a
                href="/hero/dc-batman"
              >
                 Mas...
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="col"
    >
      <div
        class="card"
      >
        <div
          class="row no-gutters"
        >
          <div
            class="col-4"
          >
            <img
              alt="Superman"
              class="card-img"
              src="/assets/dc-superman.jpg"
            />
          </div>
          <div
            class="col-8"
          >
            <div
              class="card-body"
            >
              <h5
                class="card-title"
              >
                Superman
              </h5>
              <p
                class="card-text"
              >
                Kal-El
              </p>
              <p
                class="card-text"
              >
                <small
                  class="text-muted"
                >
                  Action Comics #1
                </small>
              </p>
              <a
                href="/hero/dc-superman"
              >
                 Mas...
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="col"
    >
      <div
        class="card"
      >
        <div
          class="row no-gutters"
        >
          <div
            class="col-4"
          >
            <img
              alt="Flash"
              class="card-img"
              src="/assets/dc-flash.jpg"
            />
          </div>
          <div
            class="col-8"
          >
            <div
              class="card-body"
            >
              <h5
                class="card-title"
              >
                Flash
              </h5>
              <p
                class="card-text"
              >
                Jay Garrick
              </p>
              <p>
                Jay Garrick, Barry Allen, Wally West, Bart Allen
              </p>
              <p
                class="card-text"
              >
                <small
                  class="text-muted"
                >
                  Flash Comics #1
                </small>
              </p>
              <a
                href="/hero/dc-flash"
              >
                 Mas...
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="col"
    >
      <div
        class="card"
      >
        <div
          class="row no-gutters"
        >
          <div
            class="col-4"
          >
            <img
              alt="Green Lantern"
              class="card-img"
              src="/assets/dc-green.jpg"
            />
          </div>
          <div
            class="col-8"
          >
            <div
              class="card-body"
            >
              <h5
                class="card-title"
              >
                Green Lantern
              </h5>
              <p
                class="card-text"
              >
                Alan Scott
              </p>
              <p>
                Alan Scott, Hal Jordan, Guy Gardner, John Stewart, Kyle Raynor, Jade, Sinestro, Simon Baz
              </p>
              <p
                class="card-text"
              >
                <small
                  class="text-muted"
                >
                  All-American Comics #16
                </small>
              </p>
              <a
                href="/hero/dc-green"
              >
                 Mas...
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="col"
    >
      <div
        class="card"
      >
        <div
          class="row no-gutters"
        >
          <div
            class="col-4"
          >
            <img
              alt="Green Arrow"
              class="card-img"
              src="/assets/dc-arrow.jpg"
            />
          </div>
          <div
            class="col-8"
          >
            <div
              class="card-body"
            >
              <h5
                class="card-title"
              >
                Green Arrow
              </h5>
              <p
                class="card-text"
              >
                Oliver Queen
              </p>
              <p
                class="card-text"
              >
                <small
                  class="text-muted"
                >
                  More Fun Comics #73
                </small>
              </p>
              <a
                href="/hero/dc-arrow"
              >
                 Mas...
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="col"
    >
      <div
        class="card"
      >
        <div
          class="row no-gutters"
        >
          <div
            class="col-4"
          >
            <img
              alt="Wonder Woman"
              class="card-img"
              src="/assets/dc-wonder.jpg"
            />
          </div>
          <div
            class="col-8"
          >
            <div
              class="card-body"
            >
              <h5
                class="card-title"
              >
                Wonder Woman
              </h5>
              <p
                class="card-text"
              >
                Princess Diana
              </p>
              <p
                class="card-text"
              >
                <small
                  class="text-muted"
                >
                  All Star Comics #8
                </small>
              </p>
              <a
                href="/hero/dc-wonder"
              >
                 Mas...
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="col"
    >
      <div
        class="card"
      >
        <div
          class="row no-gutters"
        >
          <div
            class="col-4"
          >
            <img
              alt="Martian Manhunter"
              class="card-img"
              src="/assets/dc-martian.jpg"
            />
          </div>
          <div
            class="col-8"
          >
            <div
              class="card-body"
            >
              <h5
                class="card-title"
              >
                Martian Manhunter
              </h5>
              <p
                class="card-text"
              >
                J'onn J'onzz
              </p>
              <p>
                Martian Manhunter
              </p>
              <p
                class="card-text"
              >
                <small
                  class="text-muted"
                >
                  Detective Comics #225
                </small>
              </p>
              <a
                href="/hero/dc-martian"
              >
                 Mas...
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="col"
    >
      <div
        class="card"
      >
        <div
          class="row no-gutters"
        >
          <div
            class="col-4"
          >
            <img
              alt="Robin/Nightwing"
              class="card-img"
              src="/assets/dc-robin.jpg"
            />
          </div>
          <div
            class="col-8"
          >
            <div
              class="card-body"
            >
              <h5
                class="card-title"
              >
                Robin/Nightwing
              </h5>
              <p
                class="card-text"
              >
                Dick Grayson
              </p>
              <p
                class="card-text"
              >
                <small
                  class="text-muted"
                >
                  Detective Comics #38
                </small>
              </p>
              <a
                href="/hero/dc-robin"
              >
                 Mas...
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="col"
    >
      <div
        class="card"
      >
        <div
          class="row no-gutters"
        >
          <div
            class="col-4"
          >
            <img
              alt="Blue Beetle"
              class="card-img"
              src="/assets/dc-blue.jpg"
            />
          </div>
          <div
            class="col-8"
          >
            <div
              class="card-body"
            >
              <h5
                class="card-title"
              >
                Blue Beetle
              </h5>
              <p
                class="card-text"
              >
                Dan Garret
              </p>
              <p>
                Dan Garret, Ted Kord, Jaime Reyes
              </p>
              <p
                class="card-text"
              >
                <small
                  class="text-muted"
                >
                  Mystery Men Comics #1
                </small>
              </p>
              <a
                href="/hero/dc-blue"
              >
                 Mas...
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="col"
    >
      <div
        class="card"
      >
        <div
          class="row no-gutters"
        >
          <div
            class="col-4"
          >
            <img
              alt="Black Canary"
              class="card-img"
              src="/assets/dc-black.jpg"
            />
          </div>
          <div
            class="col-8"
          >
            <div
              class="card-body"
            >
              <h5
                class="card-title"
              >
                Black Canary
              </h5>
              <p
                class="card-text"
              >
                Dinah Drake
              </p>
              <p>
                Dinah Drake, Dinah Lance
              </p>
              <p
                class="card-text"
              >
                <small
                  class="text-muted"
                >
                  Flash Comics #86
                </small>
              </p>
              <a
                href="/hero/dc-black"
              >
                 Mas...
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
